faker/providers/address/__init__.py

```diff
@@ -93,8 +93,10 @@ def geo_coordinate(cls, center=None, radius=0.001):
         if not center:
             return Decimal(str(random.randint(-180000000, 180000000) / 1000000.0)).quantize(Decimal('.000001'))
         else:
+            center = float(center)
+            radius = float(radius)
             geo = random.uniform(center - radius, center + radius)
```

Comment:
We have some _basic_ testing, where every provider gets called at least once with the default arguments, but that obviously doesn't cover most cases.

New per-method tests would be a nice addition and are really appreciated.